### ISSUE TITLE
First provisioning of NLB ignores logging and cross-zone annotations

### ISSUE THREAD

**Issue by aSapien** [kubernetes/cloud-provider-aws#180]
**What happened**:
When NLB is first provisioned, S3 Logging and Cross Zone annotations are ignored.

**What you expected to happen**:
For the Logging and Cross-Zone annotations to apply on initial creation of NLB when these annotations exist:
```yaml
service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: 'true'
service.beta.kubernetes.io/aws-load-balancer-access-log-enabled: 'true'
service.beta.kubernetes.io/aws-load-balancer-access-log-s3-bucket-name: my-bucket
service.beta.kubernetes.io/aws-load-balancer-access-log-s3-bucket-prefix: my-prefix
```

**How to reproduce it (as minimally and precisely as possible)**:
1. __Create a dummy service with the relevant loadbalancer annotations. Example:__
```yaml
kind: Service
apiVersion: v1
metadata:
  name: some-service
  namespace: default
  labels:
    test: "true"
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-access-log-emit-interval: '5'
    service.beta.kubernetes.io/aws-load-balancer-access-log-enabled: 'true'
    service.beta.kubernetes.io/aws-load-balancer-access-log-s3-bucket-name: <some-bucket>
    service.beta.kubernetes.io/aws-load-balancer-access-log-s3-bucket-prefix: <some-prefix>
    service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp
    service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: 'true'
    service.beta.kubernetes.io/aws-load-balancer-type: nlb
spec:
  ports:
    - name: http
      protocol: TCP
      port: 80
      targetPort: tohttps
      nodePort: 30242
    - name: https
      protocol: TCP
      port: 443
      targetPort: http
      nodePort: 31043
  type: LoadBalancer
  externalTrafficPolicy: Local
```
2. __Check the AWS console for the attributes of the new provisioned loadbalancer. Notice that neither logging or cross-zone is applied:__

![image](https://user-images.githubusercontent.com/8168970/108717206-abb5f300-7525-11eb-936f-152af980cd53.png)

**Anything else we need to know?**:
When these annotations are added after the NLB has already been provisioned - they are applied correctly.

**Environment**: AWS/EKS 1.18
- Kubernetes version (use `kubectl version`): 1.18
- Cloud provider or hardware configuration: AWS/EKS
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/kind bug

**Comment by ayberk** [comment]
Hi @aSapien, currently it's not possible to enable the external cloud-provider on an EKS cluster. So this would be a bug on the upstream k/k.

That being said, this might be an issue we need to deal with during migration. Can you describe your setup? Are you running the kubelet with `--cloud-provider=external`?

**Comment by aSapien** [comment]
Hey @ayberk, I'm not sure I'm following. Doesn't EKS use the AWS cloud provider internally?

I can see that the annotations are parsed and handled here, and seems like the ones in question are only handled in the "UpdateLoadBalancer" flow.

Tracing the call chain, this is what I see:

- `reconcileLBAttributes` is handling the annotations:
https://github.com/kubernetes/cloud-provider-aws/blob/6c10f4b1ba2d0377665cbaa7a2cbe6730591ec64/pkg/providers/v1/aws_loadbalancer.go#L397-L413

- Which is only called from `ensureLoadBalancerv2`:
https://github.com/kubernetes/cloud-provider-aws/blob/6c10f4b1ba2d0377665cbaa7a2cbe6730591ec64/pkg/providers/v1/aws_loadbalancer.go#L135

- Which is called from the public `EnsureLoadBalancer`:
https://github.com/kubernetes/cloud-provider-aws/blob/6c10f4b1ba2d0377665cbaa7a2cbe6730591ec64/pkg/providers/v1/aws.go#L3783

- Which is called **only** from `UpdateLoadBalancer`:
https://github.com/kubernetes/cloud-provider-aws/blob/6c10f4b1ba2d0377665cbaa7a2cbe6730591ec64/pkg/providers/v1/aws.go#L4632

My setup is an EKS 1.18 cluster on AWS. 
I'm using Terraform to provision my infra, and helm-charts for K8S deployments.
For ingress, I'm using [`ingress-nginx`](https://github.com/kubernetes/ingress-nginx) helm chart with an NLB SSL termination in front of it.

My full `ingress-nginx`  helm-chart values:
```yaml
controller:
  containerPort:
    http: 80
    https: 443
    tohttps: 2443
  config:
    http-snipet: |
      server {
        listen 2443;
        return 308 https://$host$request_uri;
      }
  service:
    externalTrafficPolicy: Local
    targetPorts:
      http: tohttps
      https: http
    annotations:
      service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
      service.beta.kubernetes.io/aws-load-balancer-internal: "true"
      service.beta.kubernetes.io/aws-load-balancer-type: nlb
      service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "https"
      service.beta.kubernetes.io/aws-load-balancer-ssl-cert: ${ tls_cert_arn }
      service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp

      # The annotations below must applied __after__ the load balancer already exists

      service.beta.kubernetes.io/aws-load-balancer-access-log-enabled: "true"
      service.beta.kubernetes.io/aws-load-balancer-access-log-s3-bucket-name: ${ nlb_accesslog_bucket_name }
      service.beta.kubernetes.io/aws-load-balancer-access-log-s3-bucket-prefix: ${ nlb_accesslog_bucket_prefix }
      service.beta.kubernetes.io/aws-load-balancer-access-log-emit-interval: 5
      service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
```

**Comment by aSapien** [comment]
@ayberk I see it's been fixed in k/k 1.20 https://github.com/kubernetes/kubernetes/pull/95247

**Comment by ayberk** [comment]
> I'm not sure I'm following. Doesn't EKS use the AWS cloud provider internally?

Not yet. External cloud provider (this repo) is not enabled on EKS clusters yet, and there's no way to enable it. Therefore all EKS clusters still use the [in-tree legacy providers](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/legacy-cloud-providers/aws).

Appreciate the investigation! I'll keep this issue open because we probably need to port the fix here as well.
